include/zephyr/net/wifi_mgmt.h

```diff
@@ -144,6 +171,66 @@ struct wifi_iface_status {
 	int rssi;
 };
 
+struct wifi_ps_params {
+	enum wifi_ps enabled;
+};
+
+struct wifi_ps_mode_params {
+	enum wifi_ps_mode mode;
+};
+
+struct wifi_twt_params {
+	enum wifi_twt_operation operation;
+	enum wifi_twt_negotiation_type negotiation_type;
+	enum wifi_twt_setup_cmd setup_cmd;
+	/* Map requests to responses */
+	uint8_t dialog_token;
+	/* Map setup with teardown */
+	uint8_t flow_id;
+	union {
+		struct {
+			bool responder;
+			bool trigger;
+			bool implicit;
+			bool announce;
+			/* Wake up time */
+			uint8_t twt_wake_interval_ms;
+			/* Interval = Wake up time + Sleeping time */
+			uint32_t twt_interval_ms;
+		} setup;
+		struct teardown {
+			/* Only for Teardown */
+			bool teardown_all;
+		} teardown;
+	} u;
```

Comment:
Couldn't this be an anonymous union? The structures already have descriptive names so there should be no confusion, and IMO it'd look better in the code just to skip the union name instead of having to use a meaningless one.